reframe/core/buildsystems.py

```diff
@@ -826,7 +826,8 @@ class Spack(BuildSystem):
     #: ReFrame looks for environments in the test's
     #: :attr:`~reframe.core.pipeline.RegressionTest.sourcesdir`.
```

Comment:
Also, I don't think this is true, as far as I can tell the environment is looked in the stage directory

Reply:
The contents of the `sourcesdir` are copied over to the stage directory before entering the compile stage. The environment is a test "resource", so it has to reside in the `sourcesdir` if given.

Reply:
```console
% spack env create -d tutorials/basics/stream/spack-env
==> Updating view at /home/mose/repo/reframe/tutorials/basics/stream/spack-env/.spack-env/view
==> Created environment in /home/mose/repo/reframe/tutorials/basics/stream/spack-env
==> You can activate this environment with:
==>   spack env activate /home/mose/repo/reframe/tutorials/basics/stream/spack-env
% ./bin/reframe -c tutorials/basics/stream/stream1.py -r --performance-report
[ReFrame Setup]
  version:           3.7.0-dev.2+1bc108c6
  command:           './bin/reframe -c tutorials/basics/stream/stream1.py -r --performance-report'
  launched by:       mose@dream
  working directory: '/home/mose/repo/reframe'
  settings file:     '/home/mose/repo/reframe/tutorials/config/mysettings.py'
  check search path: '/home/mose/repo/reframe/tutorials/basics/stream/stream1.py'
  stage directory:   '/home/mose/repo/reframe/stage'
  output directory:  '/home/mose/repo/reframe/output'

[==========] Running 1 check(s)
[==========] Started on Thu Jun 24 23:28:26 2021 

[----------] started processing Test (Test)
[ RUN      ] Test on catalina:default using gnu
[     FAIL ] (1/1) Test on catalina:default using gnu [compile: 0.784s run: n/a total: 0.793s]
==> test failed during 'compile_wait': test staged in '/home/mose/repo/reframe/stage/catalina/default/gnu/Test'
[----------] finished processing Test (Test)

[----------] waiting for spawned checks to finish
[----------] all spawned checks have finished

[  FAILED  ] Ran 1/1 test case(s) from 1 check(s) (1 failure(s), 0 skipped)
[==========] Finished on Thu Jun 24 23:28:27 2021 

==============================================================================
SUMMARY OF FAILURES
------------------------------------------------------------------------------
FAILURE INFO for Test 
  * Test Description: Test
  * System partition: catalina:default
  * Environment: gnu
  * Stage directory: /home/mose/repo/reframe/stage/catalina/default/gnu/Test
  * Node list: 
  * Job type: local (id=None)
  * Dependencies (conceptual): []
  * Dependencies (actual): []
  * Maintainers: []
  * Failing phase: compile_wait
  * Rerun with '-n Test -p gnu --system catalina:default -r'
  * Reason: build error: stdout: 'rfm_Test_build.out', stderr: 'rfm_Test_build.err'
--- rfm_Test_build.err (first 10 lines) ---
==> Error: No such environment: 'spack-env'
==> Error: `spack add` requires an environment
  activate an environment first:
      spack env activate ENV
  or use:
      spack -e ENV add ...
--- rfm_Test_build.err --- 
------------------------------------------------------------------------------

Log file(s) saved in: '/tmp/rfm-rg3uuekm.log'
```
Either I'm doing something wrong (totally possible but documentation would help :slightly_smiling_face:) or that doesn't work

Reply:
Oooh, sorry, now I see: I need to set `sourcesdir` in the test.  Well, I think this is one more reason for this pull request :slightly_smiling_face: